plugins/catalog-backend/src/database/DatabaseManager.ts

```diff
@@ -44,6 +44,7 @@ export class DatabaseManager {
   ): Promise<Database> {
     await knex.migrate.latest({
       directory: migrationsDir,
+      tableName: 'knex_migrations_catalog',
```

Comment:
I don't think we want to set these, and instead provided different databases for each backend plugin. Using this to keep migrations apart would still put all tables in the same namespace, which imo doesn't provide enough isolation

Reply:
Could we support both? As in, keeping this but still strongly suggest to adopters that databases still should be separate (but we don't fail in weird ways if you choose to keep them together)?

Reply:
Maybe we can provide a way to automatically create missing DBs? And for databases that support multiple schemas in a DB, allow schemas as an option to namespace things as well?

I also think it's a bit unfortunate that we end up with a plugin ID inside the plugin itself for this case, afaik all references to the plugin ID are in the backend atm. Ideally they'd only exist in one of the two, or it just gets unnecessarily tricky to change.

Reply:
>  which imo doesn't provide enough isolation

What is the risk @Rugvip? Are you worried about resilience? If one DB goes down, all of Backstage is gone for that company.

Reply:
A couple of things. Main one being that plugins would be tempted to read from each other's tables, creating an architectural nightmare. There's also the risk of simple naming collision of tables between plugins.

I also think it will be very handy if you can easily dump and restore plugin DBs as a method of moving plugins around in a deployment. One main idea we have to the backend architecture is to start small and simple and break things up as you grow. That probably means having a couple of backend plugins in a deployment to begin with, and over time splitting them out to multiple deployments. That process should be a lot easier if there's clear separation between plugin DBs.

Reply:
That makes a lot of sense as a default. I suspect there will be many cases where companies will want to have multiple bespoke plugins referencing the same DB so it might be best to support both cases.

Reply:
Multiple microservices talking to a single DB and

<img src="https://user-images.githubusercontent.com/4984472/86902328-1983ea80-c10e-11ea-80e6-987fd3ed63dc.png" height="200"/>

Tbh I wouldn't want to encourage that pattern at all, at most allow it 😁

Reply:
By choosing to have a separate DB for each plugin, what you're essentially doing is allowing the plugins to define their own database configurations. Anyone can then use two plugins with the same database configuration thus mimicking the single database scenario. So, I think that is the better choice, even if it put more burden to the deployer to configure all the needed plugins to point to the same DB.

Another reason you may want to do that, is to disallow plugins reading into secrets stored on the database by other plugins, thus separating sensitive data on that level in a simple way.

Of course, plugins or "services" should be autonomous and thus should be able to be added, removed, or moved as a single coherent thing, not as something that depends on data that others (other plugins/services) provided.